core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -302,6 +302,7 @@ public T acquire() throws IOException {
    * @param unit the unit to use for time
    * @return a resource taken from the pool
    * @throws TimeoutException if it fails to acquire because of time out
+   * @throws IOException if the thread is interrupted while acquiring the resource
```

Comment:
Why not to throw `InterruptedException` directly out of this method? That sounds natural for a blocking method.

Reply:
You have a good point, and I think that is the right approach. This was a quick patch and we would like to follow up with refactoring in the master branch. See https://github.com/Alluxio/alluxio/issues/9679 
Adding `InterruptedException` would have impacted a large part of the codebase which is why we settled on this implementation.